.github/workflows/ci-docs-autogen.yml

```diff
@@ -0,0 +1,74 @@
+name: Autogenerated documentation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  autogen-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: yarn
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Node dependencies cache
+        uses: actions/cache@v3
+        with:
+          path: "node_modules"
+          key: yarn-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+        shell: bash
+
+      - name: Generate docs
+        run: forge doc --build
+        shell: bash
+
+      - name: Upload docs
+        uses: actions/upload-artifact@v2
+        with:
+          name: docs-foundry
+          path: docs/book
+
+  upload-to-s3:
+    runs-on: ubuntu-latest
+    needs: autogen-docs
+    environment:
+      name: docs
+      url: https://developers.morpho.xyz
+    steps:
+      - name: Download docs
+        uses: actions/download-artifact@v2
+        with:
+          name: docs-foundry
+          path: docs
+
+      - name: Setup AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Comment:
Were these created specifically for this CI/CD? Are you able to identify them anytime to react in case they are exposed (are they well named in AWS)?

Reply:
- They are generated for this purpose only, scoped to minimal resources and monitored in case of activity (through CloudTrail).

Ping me if you have interesting good practices about aws credentials monitoring ;)